### PR TITLE
feat: team details page refactor and tests

### LINKED ISSUE #1172: [FEATURE]: team details page refactor and tests
## Summary

- Refactor team details page layout
- Add tests for the team details page

### PR DESCRIPTION
**Pull request by JoaoSaIvador**
Relates to #1172 
Fixes #1172

## Proposed Changes

  - Refactor the components related to the team details page
  - Add tests to the components related to the team details page
  - Add the components related to the team details page to Storybook

This pull request closes #1172

### CODE REVIEW

File: frontend/src/utils/constants.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ export const REFRESH_TOKEN_ERROR = 'REFRESH_TOKEN_ERROR';
 
 export const MIN_MEMBERS = 4;
 
-export const RECOIL_DEV_TOOLS = NEXT_PUBLIC_RECOIL_DEV_TOOLS === 'true';
+export const RECOIL_DEV_TOOLS = NEXT_PUBLIC_RECOIL_DEV_TOOLS !== 'true';
```
Comment (nunocaseiro) [inline frontend/src/utils/constants.ts:36]: Why is this changed?

**Review by nunocaseiro — APPROVED** [review]

File: frontend/src/utils/constants.ts (hunk 1 above)
Comment (JoaoSaIvador) [reply frontend/src/utils/constants.ts:36]: Sorry, we were testing something to do with the recoil dev tools and left that behind by accident

**Review by nunocaseiro — APPROVED** [review]
